spring-pulsar-test/src/test/java/org/springframework/pulsar/test/support/PulsarConsumerTestUtilTests.java

```diff
@@ -97,9 +115,27 @@ void exceptionIsThrownWhenConditionNotMetWithinAwaitDuration() {
 			.withMessage("Condition was not met within 5 seconds");
 	}
 
+	@Test
+	void exceptionIsThrownWhenMultipleUntilConditionsAreChainedTogether() {
```

Comment:
**[TinyNit]** I believe an exception is thrown when until is called multiple times.